src/main.cpp

```diff
@@ -3654,16 +3649,24 @@ bool AcceptBlockHeader(const CBlockHeader &block,
         if (pindexPrev->nStatus & BLOCK_FAILED_MASK)
             return state.DoS(100, error("%s: previous block invalid", __func__), REJECT_INVALID, "bad-prevblk");
 
+        // If the parent block is not the checkpointed block, then we are on the wrong fork so ignore.
```

Comment:
Isn't this comment misleading?

`CheckAgainstCheckpoint` returns true if block height of the block we are checking doesn't belong to set of keys of `checkpointData` and this is true for the large majority of the blocks. 

In case the block `height `belongs to the `checkpointData` keys set, then we check if the block we examining has the hash as the one defined `checkpointData` map.

if this is not the case then we return false and we mark as invalid the chain the block belongs to.

I would used something like:

    If the parent block height happens to belong to the set of check-pointed
    blocks but it has a mismatched hash, then we are on the wrong fork so ignore